omegaconf/listconfig.py

```diff
@@ -390,13 +390,22 @@ def _get_node(
             if value is not None:
                 if isinstance(key, slice):
                     assert isinstance(value, list)
-                    for v in value:
+                    indices = range(len(self))[key]
+                    for idx, v in zip(indices, value):
                         if throw_on_missing_value and v._is_missing():
-                            raise MissingMandatoryValue("Missing mandatory value")
+                            self._format_and_raise(
+                                key=idx,
+                                value=None,
+                                cause=MissingMandatoryValue("Missing mandatory value"),
+                            )
```

Comment:
This is a lot of custom logic for pretty low return.
I can live with a simpler handling for slice errors with missing (no need to report the exact first missing key).
I think just use the slice as is without reporting the exact index. it would also simplify your tests.

Reply:
Done in https://github.com/omry/omegaconf/pull/503/commits/01d53955351a9db389cf7ac3ee42b3f0085c081e